src/hooks/useUAuthUser.ts

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+import UAuth from '@uauth/js'
+import { UserInfo } from '@uauth/js'
+import { UAUTH_CONFIG } from '../connectors'
+
+export default function useUAuthUser(walletAddress?: string) {
+  const [user, setUser] = React.useState<UserInfo | undefined>(undefined)
+
+  React.useEffect(() => {
+    async function fetchUAuthUser() {
+      const uauth = new UAuth(UAUTH_CONFIG)
+
+      try {
+        console.log('uauthUser before get')
+        console.log('walletAddress', walletAddress)
+        const uauthUser = await uauth.user()
+        if (walletAddress && walletAddress.toLowerCase() === uauthUser.wallet_address?.toLowerCase()) {
+          setUser(uauthUser)
+        } else {
+          setUser(undefined)
+        }
+        console.log('uauthUser', uauthUser)
+      } catch (error) {
+        console.log('uauthUser error', error)
+        setUser(undefined)
```

Comment:
May we can remove the console.logs here?

Reply:
Done, thanks for the recommendations!

Reply:
Nice